crates/sui-framework-snapshot/src/lib.rs

```diff
@@ -128,3 +127,33 @@ pub fn load_bytecode_snapshot(protocol_version: u64) -> anyhow::Result<Vec<Syste
 pub fn manifest_path() -> PathBuf {
     PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("manifest.json")
 }
+
+/// Given a protocol version:
+/// * The path to the snapshot directory for that version is returned, if it exists.
+/// * If the version is greater than the latest snapshot version, then `Ok(None)` is returned.
+/// * If the version does not exist, but there are snapshots present with versions greater than
+///   `version`, then the smallest snapshot number greater than `version` is returned.
+fn snapshot_path_for_version(version: u64) -> anyhow::Result<PathBuf> {
+    let snapshot_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("bytecode_snapshot");
+    let mut snapshots = BTreeSet::new();
+
+    for entry in fs::read_dir(&snapshot_dir)? {
+        let entry = entry?;
+        let path = entry.path();
+        if path.is_dir() {
+            if let Some(snapshot_number) = path
+                .file_name()
+                .and_then(|n| n.to_str())
+                .and_then(|n| n.parse::<u64>().ok())
+            {
+                snapshots.insert(snapshot_number);
+            }
+        }
+    }
+
+    snapshots
```

Comment:
I think you probably want the largest version that is <= the current version, instead of the smallest version that is >=. See `sui::framework_versions::version_for_protocol`